### ISSUE TITLE
class diagrams: allow notes for namespace'd classes

### ISSUE THREAD

**Issue by The-Alchemist** [mermaid-js/mermaid#4706]
## Short Description

...should improve its support for notes inside namespaces.

## Example Syntax

The following example shows a note for a specific class _inside_ a namespace.  This syntax currently results in  a parse error, `Expecting CLASS, got NOTE_FOR`:

```mermaid

classDiagram
namespace BaseShapes {

    note for Triangle "i am a note for Triangle"
    class Triangle
    class Rectangle {
      double width
      double height
    }
}
```

You can [play with this syntax here on the playground](https://mermaid.live/edit#pako:eNplkMFuhCAQhl-FzNkYxSIux6bXXro9NVymMiqJglFMuzW-e1E3adOSSWA--P-ZYYXaGwIFdY_z_GSxnXDQzuFA84g1sUec6drhSDNbtdOOxeV8INb4ib1OFl3bE9NgGQ4M_19pODWH_w_9xV6oDqfLemLGjF_eY_5hTej-sI5s24UTbtrFgAQGmga0Jo5xWGgIHQ2xtIpHQw0ufdj72OJTXIK_3lwNKkwLJbCMBgPdBwfVYD9HOqIDtcInKJ5nKZc554UQXJQPokrgBiqXZSrLC5fFRcqiFLncEvjyPlpkqahElUleFpXI83JXkLHBT8_3r963o8bbIdgb2b4B0sF8Pg).

## Details
Technically, you can add a note for a namespaced class, but you have to place the note _outside_ the namespace, and therefore, it renders _outside_ the namespace.

```mermaid
classDiagram
namespace BaseShapes {

    class Triangle
    class Rectangle {
      double width
      double height
    }
}
note for Triangle "i am a note for Triangle"

```

<note> The above diagram doesn't render in Github's viewer, but [you can view it on the playground](https://mermaid.live/edit#pako:eNplkE1vhCAQhv8KmbMxikVcjptee-n21HCZyriSKBjFtFvjfy_STdO0k0mAZ975YoPWGwIF7YDL8mjxOuOoncORlglbYmdc6NLjRAvbtNOORUta9jJbdNeBfrNnakOCh5glM359i-93a0L_h_Vkr334hrt20Z0PxDo__xRnGizDkSH7F9KgHWQw0jyiNXGF1FJD6GmMQRWvhjpch3Ao9yjFNfjLzbWgwrxSButkMNB9aVAdDkukEzpQG3yA4mWRc1lyXgnBRf0gmgxuoEpZ57I-cVmdpKxqUco9g0_vY4kiF41oCsnrqhFlWR8ZZGzw89P9m48j9XhNCccg-xdnLnu-).  It appears like this:
![image](https://github.com/mermaid-js/mermaid/assets/812287/a85c3ae6-a825-4e9e-b8f7-7621f5ec7dc1)

### Example

```

classDiagram
namespace BaseShapes {

    note for Triangle "i am a note for Triangle"
    class Triangle
    class Rectangle {
      double width
      double height
    }
}
```

**Comment by Adarsha186** [comment]
Hi @The-Alchemist, is this issue still open, I'm interested in doing open source contributions, if this issue is open can I take up this task.

**Comment by The-Alchemist** [comment]
> Hi @The-Alchemist, is this issue still open, I'm interested in doing open source contributions, if this issue is open can I take up this task.

I think @kairi003 is working on it, or is done.  See !5814

But please don't hesitate to pick up another task!

**Comment by kairi003** [comment]
@The-Alchemist Yes, the pr to resolve this issue is done. Now I am waiting for the merge, is there anything I can do for that?

**Comment by jgreywolf** [comment]
I am rerunning the tests to see if we get past the timeout failure there

**Comment by kairi003** [comment]
> Error: You have reached the maximum screenshot capacity included in your Pro open source xl Plan. Please upgrade your Plan.
https://github.com/marketplace/argos-ci
![image](https://github.com/user-attachments/assets/c2293140-0118-4e90-9f65-33496bcbf41d)

@jgreywolf I tried re-running as well, but to no avail.
The message says _reached the maximum screenshot capacity_, but mermaid is already on the highest plan: XL.
I think we need to wait until December for the capacity to reset in order to pass the e2e tests.

**Comment by jgreywolf** [comment]
@sidharthv96 @ashishjain0512 @knsv @aloisklink  Can one of you assist with this?

**Comment by jgreywolf** [comment]
@darshanr0107 Are you actively looking into this?

**Comment by The-Alchemist** [comment]
https://github.com/mermaid-js/mermaid/pull/5814 is waiting for a successful build.  It's been waiting for about a year, unfortunately.

**Comment by ghost** [comment]
@Yup

**Comment by ghost** [comment]
[ülla]()

**Comment by ghost** [comment]
- []()

**Comment by kairi003** [comment]
Hi, I've fixed #5814
